Branch-SDK/src/main/java/io/branch/referral/util/BranchEvent.java

```diff
@@ -244,16 +244,48 @@ public String getEventName() {
      * Logs this BranchEvent to Branch for tracking and analytics
      *
      * @param context Current context
-     * @return {@code true} if the event is logged to Branch
+     * @return {@code true}
      */
     public boolean logEvent(Context context) {
-        boolean isReqQueued = false;
+        logEvent(context, null);
+        return true;
+    }
+
+    public interface BranchLogEventCallback {
+        void onEventLogged(boolean success);
```

Comment:
Given the names `onEventLogged` and `onFailure` I think it's fair to assume the boolean you pass in `onEventLogged` is always true. I'd rename as `onSuccess` to be consistent with other callback apis.

Reply:
It *is* possible we fail to queue it 😅